pkg/resource/parameter_group/hooks.go

```diff
@@ -158,6 +159,119 @@ func (rm *resourceManager) resetAllParameters(
 	return &resource{ko}, nil
 }
 
+// customUpdate overrides sdkUpdate by custom logic
+func (rm *resourceManager) customUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdate")
+	defer func() {
+		exit(err)
+	}()
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.updateTags(ctx, desired, latest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if delta.DifferentAt("Spec.ParameterNameValues") {
+		ko, err := rm.resetParameterGroup(ctx, desired, latest)
+
+		if ko != nil || err != nil {
+			return ko, err
+		}
+	}
+
+	if !delta.DifferentExcept("Spec.Tags") {
+		return desired, nil
+	}
+
+	inputs := rm.updateRequestPayload(desired)
+
+	var resp *svcsdk.UpdateParameterGroupOutput
+	_ = resp
+
+	// Update 20 parameters each time
+	for _, input := range inputs {
+		resp, err = rm.sdkapi.UpdateParameterGroupWithContext(ctx, input)
+		rm.metrics.RecordAPICall("UPDATE", "UpdateParameterGroup", err)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.ParameterGroup.ARN != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.ParameterGroup.ARN)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+	if resp.ParameterGroup.Description != nil {
+		ko.Spec.Description = resp.ParameterGroup.Description
+	} else {
+		ko.Spec.Description = nil
+	}
+	if resp.ParameterGroup.Family != nil {
+		ko.Spec.Family = resp.ParameterGroup.Family
```

Comment:
This is going to throw an error if `resp` is nil. Also is `resp.ParameterGroup.Family` going to be the same for every request?

Reply:
A `parameter group` resource only has one value to `Family` field. If these multiple requests are all for this specific resource, then `Family` field should always be the same value. 
I will add an empty `UpdateParameterGroupWithContext` call to make sure `resp` is not `nil`.

Reply:
I'd rather we don't call `UpdateParameterGroupWithContext` unnecessarily. We could hit limits more quickly if we call it more than we need to. Instead, we can just check `if resp != nil {}` before we make any changes that require `resp`

Reply:
Modified.